### ISSUE TITLE
[Bug] Undirected.Graph.toAdjacencyMatrix only shows an edge one time instead 

### ISSUE THREAD

**Issue by LibraChris** [CSBiology/FSharp.FGL#17]
### Description

Graph.toAdjacencyMatrix should show an edge on both connected vertices. But in this case the edge only shows up once.

### Example Graph

open Aether
open FSharp.FGL 
open FSharp.FGL.Undirected

Graph.empty
|> Vertices.addMany [0,1; 1,1; 2,1; 3,1; 4,1; 5,2; 6,2; 7,2; 8,2]
|> Undirected.Edges.addMany [0,0,1; 0,1,1; 0,3,1; 0,4,1; 1,2,1; 1,4,1; 2,3,1; 3,4,1; 2,5,1; 5,6,1; 5,7,1; 5,8,1; 6,7,1; 7,8,1]
|>Graph.toAdjacencyMatrix

### Solution
Already solved, pull request follows soon.

**Comment by HLWeil** [comment]
Could you write a sentence or two about the outcome you observed and the solution your proposed in your PR?

**Comment by LibraChris** [comment]
### Observed Outcome
For the example above the observed outcome is:
```
val it : int [] [] =
  [|[|1; 1; 0; 1; 1; 0; 0; 0; 0|]; [|0; 0; 1; 0; 1; 0; 0; 0; 0|];
    [|0; 0; 0; 1; 0; 1; 0; 0; 0|]; [|0; 0; 0; 0; 1; 0; 0; 0; 0|];
    [|0; 0; 0; 0; 0; 0; 0; 0; 0|]; [|0; 0; 0; 0; 0; 0; 1; 1; 1|];
    [|0; 0; 0; 0; 0; 0; 0; 1; 0|]; [|0; 0; 0; 0; 0; 0; 0; 0; 1|];
    [|0; 0; 0; 0; 0; 0; 0; 0; 0|]|]
```
### Expected Outcome
For the example above the expected outcome would be:
```
val it : int [] [] =
  [|[|1; 1; 0; 1; 1; 0; 0; 0; 0|]; [|1; 0; 1; 0; 1; 0; 0; 0; 0|];
    [|0; 1; 0; 1; 0; 1; 0; 0; 0|]; [|1; 0; 1; 0; 1; 0; 0; 0; 0|];
    [|1; 1; 0; 1; 0; 0; 0; 0; 0|]; [|0; 0; 1; 0; 0; 0; 1; 1; 1|];
    [|0; 0; 0; 0; 0; 1; 0; 1; 0|]; [|0; 0; 0; 0; 0; 1; 1; 0; 1|];
    [|0; 0; 0; 0; 0; 1; 0; 1; 0|]|]
```

### Solution
I propose minor additions to the very last functions of the code shown below :

from
```
    ///Transforms a graph into an adjacency matrix of its edges.
    let inline toAdjacencyMatrix (g:Graph<'Vertex,'Label,'Edge>) =
        //Create a hashmap of the vertices
        let hashMap = System.Collections.Generic.Dictionary<'Vertex,int>()
        let n = 
            let rec loop g i= 
                match Graph.decomposeFirst g with
                | Some (_,v,_,_),g -> 
                    hashMap.[v] <- i
                    loop g (i+1)        
                | None, _ -> i
            loop g 0
        //Create the matrix
        let adj : 'Edge [][] = Array.init n (fun _ -> 
            Array.zeroCreate n)
        //Fill the matrix with values by using the hashmap as an index finder
        Edges.iter (fun v1 v2 e -> adj.[hashMap.Item v1].[hashMap.Item v2] <- e) g
        adj

    ///Transfroms a graph into a adjacency matrix, maps every edge using the projection.
    let inline toAdjacencyMatrixBy (projection : 'Edge -> 'REdge) (g:Graph<'Vertex,'Label,'Edge>) =
        //Create a hashmap of the vertices
        let hashMap = System.Collections.Generic.Dictionary<'Vertex,int>()
        let n = 
            let rec loop g i= 
                match Graph.decomposeFirst g with
                | Some (_,v,_,_),g -> 
                    hashMap.[v] <- i
                    loop g (i+1)        
                | None, _ -> i+1
            loop g 0
        //Create the matrix
        let adj : 'REdge [][] = Array.init n (fun _ -> 
            Array.zeroCreate n)
        //Fill the matrix with values by using the hashmap as an index finder
        Edges.iter (fun v1 v2 e -> adj.[hashMap.Item v1].[hashMap.Item v2] <- projection e) g
        adj
```

To
```
    ///Transforms a graph into a adjacency matrix of its edges. If there is no edge between two vertices, the noEdgeValue is used.
    let inline toAdjacencyMatrix (g:Graph<'Vertex,'Label,'Edge>) =
        //Create a hashmap of the vertices
        let hashMap = System.Collections.Generic.Dictionary<'Vertex,int>()
        let n = 
            let rec loop g i= 
                match Graph.decomposeFirst g with
                | Some (_,v,_,_),g -> 
                    hashMap.[v] <- i
                    loop g (i+1)        
                | None, _ -> i
            loop g 0
        //Create the matrix
        let adj : 'Edge [][] = Array.init n (fun _ -> 
            Array.zeroCreate n)
        //Fill the matrix with values by using the hashmap as an index finder
        Edges.iter (fun v1 v2 e -> 
            adj.[hashMap.Item v1].[hashMap.Item v2] <- e
            adj.[hashMap.Item v2].[hashMap.Item v1] <- e
            ) g
        adj

    ///Transfroms a graph into a adjacency matrix, maps every edge using the projection.
    let inline toAdjacencyMatrixBy (projection : 'Edge -> 'REdge) (g:Graph<'Vertex,'Label,'Edge>) =
        //Create a hashmap of the vertices
        let hashMap = System.Collections.Generic.Dictionary<'Vertex,int>()
        let n = 
            let rec loop g i= 
                match Graph.decomposeFirst g with
                | Some (_,v,_,_),g -> 
                    hashMap.[v] <- i
                    loop g (i+1)        
                | None, _ -> i+1
            loop g 0
        //Create the matrix
        let adj : 'REdge [][] = Array.init n (fun _ -> 
            Array.zeroCreate n)
        //Fill the matrix with values by using the hashmap as an index finder
        Edges.iter (fun v1 v2 e -> 
            let e' = projection e
            adj.[hashMap.Item v1].[hashMap.Item v2] <- e'
            adj.[hashMap.Item v2].[hashMap.Item v1] <- e'
            ) g
        adj
```

**Comment by HLWeil** [comment]
fixed in 43a211d
